src/ComposerRequireChecker/FileLocator/LocateComposerPackageDirectDependenciesSourceFiles.php

```diff
@@ -12,21 +19,59 @@ public function __invoke(string $composerJsonPath): Generator
 
         $composerJson = json_decode(file_get_contents($composerJsonPath), true);
         $configVendorDir = $composerJson['config']['vendor-dir'] ?? 'vendor';
-        $vendorDirs = array_values(
-            array_map(
-                function (string $vendorName) use ($packageDir, $configVendorDir) {
-                    return $packageDir . '/' . $configVendorDir . '/' . $vendorName;
-                },
-                array_keys($composerJson['require'] ?? [])
-            )
-        );
-
-        foreach ($vendorDirs as $vendorDir) {
+
+        $installedPackages = $this->getInstalledPackages($packageDir . '/' . $configVendorDir);
+
+        $vendorDirs = [];
+        foreach ($composerJson['require'] ?? [] as $vendorName => $vendorRequiredVersion) {
+            $vendorDirs[$vendorName] = $packageDir . '/' . $configVendorDir . '/' . $vendorName;
+        };
+
+        foreach ($vendorDirs as $vendorName => $vendorDir) {
+            $tmp = null;
+
+            # If this package doesn't export its composer.json file then create a temporary one from installed.json
             if (!file_exists($vendorDir . '/composer.json')) {
```

Comment:
I'd say that `installed.json` should be taken all the time. Are there disadvantages in doing so?

Reply:
I was going for least disruption. By supporting both we avoid compatibility issues with older/newer versions of composer that may not provide it.
If you'd rather keep this code simple though I'll see if I can get an answer from the composer guys on the reliability of `installed.json`

Reply:
It looks like there's an upcoming BC break in `installed.json` (https://github.com/composer/composer/pull/7999) so I think it would be more robust to try and use the vendor's `composer.json` first (BC breaks here are almost unimaginable), and only use `installed.json` as a fallback

Reply:
Yes, my plan would be to always use `installed.json`, rather than looking for the files. It's also fine to discuss BC boundaries around `installed.json` with upstream composer maintainers.

Reply:
That's indeed something that a separate dependency does (by reading the autoloader definitions).

If we can feed that information through, we can get rid of all `composer.json` access operations.

Reply:
Ok cool, shall I submit a separate PR to change that dependency to require a data structure instead of the path to `composer.json`? What about BC?

Reply:
We can bump the major version of the lib